### ISSUE TITLE
Date Picker Pager: Markers are a bit too wide

### ISSUE THREAD

**Issue by StefanOltmann** [alexzhirkevich/compose-cupertino#15]
Please match the size of the circles closer to the original.

It's not only bigger than the original, it somehow feels too big, too.

<img width="295" alt="grafik" src="https://github.com/alexzhirkevich/compose-cupertino/assets/9299303/4e245f94-b822-4ac0-9af6-a278b3b3ff95">

**Comment by alexzhirkevich** [comment]
It's filling width because it use lazy row under the hood.

This will happen otherwise:
![image](https://github.com/alexzhirkevich/compose-cupertino/assets/63979218/f0f9ebc2-a5e4-4417-8fc7-4a1a391d4a53)

**Comment by alexzhirkevich** [comment]
Looks pretty identical to me 😅. Native even 2% bigger 
<img src="https://github.com/alexzhirkevich/compose-cupertino/assets/63979218/d7f572dd-cd70-4ad4-9e2f-dbd5a5649734" width = "300">

<img src="https://github.com/alexzhirkevich/compose-cupertino/assets/63979218/795601a9-4d7b-4faf-8f81-2df3926ce4c0" width = "300">

**Comment by StefanOltmann** [comment]
Oh, okay. You are right. It's not that bigger.

I now see that it looks like that to me because your component has more spacing than the date picker I'm looking at in Apple Photos.

Original in Apple Photos:
![iphone8_apple_photos](https://github.com/alexzhirkevich/compose-cupertino/assets/9299303/1640766f-9727-4b97-8ca2-d3b3fc69883a)

CupertinoDatePicker in Ashampoo Photos:
![iphone8_ashamoo_photos](https://github.com/alexzhirkevich/compose-cupertino/assets/9299303/0861db22-8b76-41e1-8e91-d5d579ad2880)

Overlay:
<img width="358" alt="comparison" src="https://github.com/alexzhirkevich/compose-cupertino/assets/9299303/666f6b84-91a6-4945-9d77-09b2f0916667">

I guess what I'm looking for is a way to configure / adjust the hardcoded spacings to match it more closely to the one Apple Photos has.

**Comment by StefanOltmann** [comment]
How exactly did you do that?

What I tried does not look so good.

```
                            CupertinoDatePicker(
                                modifier = Modifier
//                                    .padding(16.dp)
                                    .width(310.dp - 16.dp)
                                    .border(1.dp, Color.Red),
                                state = datePickerState,
                                style = DatePickerStyle.Pager()
                            )
```

<img width="309" alt="grafik" src="https://github.com/alexzhirkevich/compose-cupertino/assets/9299303/b42ea854-d146-40fa-899a-45371fca4420">
<img width="322" alt="grafik" src="https://github.com/alexzhirkevich/compose-cupertino/assets/9299303/7e51d7fc-80f4-47c3-ab65-992300ab300b">

**Comment by StefanOltmann** [comment]
> Horizontal spacings are not hardcoded.

But a vertical spacing of `6.dp` is currently hardcoded.

**Comment by alexzhirkevich** [comment]
Seems like native date picker really makes circles smaller when width is limited. Anyway, your first screenshot without the paddings looks pretty good to me.

**Comment by StefanOltmann** [comment]
It indeed may have to do with the display zoom feature.

I design my Ashampoo Photos for iPhone SE with display zoom enabled, because this is the smallest screen size I know of that users might likely have. In this case I did not check how Apple Photos looks without it or on bigger screens. If it fits into the smallest screen, it will fit into all others.

The iPhone SE screen is 320 x 548 pt.

I would like to make the date picker width a bit smaller, but in fact it's now 310dp (7 x 42.dp buttons plus 8.dp padding on both sides) and fits into the screen.

You can't say that about the Material3 monster, which is at least 7 x 48.pt = 336.pt wide and won't fit a small screen. So I'm thankful you made this one and will use it on all platforms eventually. :)

We can drop my initial request for a smaller circle with all we know now. It's not too important.

What would really help me instead is if you could turn the `VerticalPadding` into a parameter. `6.dp` is a decent default and it looks good if you have plenty of space, but in my case I require the date picker height to be shorter, because I can't fit an element.

Without the element:
![Screenshot 2024-01-09 at 15 54 43](https://github.com/alexzhirkevich/compose-cupertino/assets/9299303/f001dd9c-1292-4ccf-813f-94241f0d22e4)

Without my missing element:
![Screenshot 2024-01-09 at 16 06 43](https://github.com/alexzhirkevich/compose-cupertino/assets/9299303/34eda373-fc09-4c84-8b02-df46be68e34b)

See?

**Comment by StefanOltmann** [comment]
Sounds great!

Maybe provide even an way to provide custom text styles similar to the way we can override the colors now.

**Comment by StefanOltmann** [comment]
> You can override it with body and headline of CupertinoTheme typography now

Can you provide me a sample how to do that?

**Comment by alexzhirkevich** [comment]
The same way you do it with material:
```kotlin
CupertinoTheme(
     typography = Typography()
){
   ...
}
```

**Comment by alexzhirkevich** [comment]
That tab control you are using at the top is also available here as `CupertinoSegmentedControl` with some nice interaction animations

**Comment by alexzhirkevich** [comment]
Available in `0.1.0-alpha03`

**Comment by StefanOltmann** [comment]
@alexzhirkevich Thank you! I've just given it a try, and the added customization options really make a significant difference, especially when working with smaller screens. :)
